frontend/src/bundles/common/components/slider/slider.tsx

```diff
@@ -0,0 +1,64 @@
+import { Box, Slider } from '@mui/material';
+import { styled } from '@mui/material/styles';
+import { useCallback } from 'react';
+
+import { getValidClassNames } from '../../helpers/helpers.js';
+import { defaultExperienceMarks } from './default-values.js';
+import classes from './style.module.css';
+import { defaultSliderStyle } from './styles.js';
+
+type Mark = {
+    label?: string;
+    value: number;
+};
+
+type Properties = {
+    label?: string;
+    value?: number;
+    className?: string;
+    step?: number | null;
+    valueLabelDisplay?: 'auto' | 'on' | 'off';
+    marks: Mark[];
+};
+
+const CustomSlider: React.FC<Properties> = ({
+    label,
+    value,
+    marks = defaultExperienceMarks,
+    className = classes.sliderContainerStyle,
+    step = null,
+    valueLabelDisplay = 'on',
+}) => {
+    const StyledSlider = styled(Slider)(defaultSliderStyle);
+
+    const getValueLabel = useCallback(
+        (value: number): string | null => {
+            return Array.isArray(marks)
+                ? (marks.find((mark) => mark.value === value)?.label as string)
+                : null;
+        },
```

Comment:
Don't use 'as string' casting in this case. Instead find value, check if this item exist and after if not return null in other case return label

Reply:
I can propose that variant
```javascript
    const getValueLabel = useCallback(
        (value: number): string | null => {
            const label = marks.find((mark) => mark.value === value)?.label;
            return label ?? null;
        },
        [marks],
    );
```

Reply:
I propose some changes in this variant in other comment